src/metabase/api/field.clj

```diff
@@ -65,16 +100,49 @@
      [:distincts (metadata/field-distinct-count field)]]))
 
 (def ^:private empty-field-values
-  {:values {} :human_readable_values {}})
+  {:values []})
 
 (api/defendpoint GET "/:id/values"
   "If `Field`'s special type derives from `type/Category`, or its base type is `type/Boolean`, return
    all distinct values of the field, and a map of human-readable values defined by the user."
   [id]
   (let [field (api/read-check Field id)]
-    (if-not (field-should-have-field-values? field)
-      empty-field-values
-      (create-field-values-if-needed! field))))
+    (if-let [field-values (and (field-should-have-field-values? field)
+                               (create-field-values-if-needed! field))]
+      (-> field-values
+          (assoc :values (field-values->pairs field-values))
+          (dissoc :human_readable_values))
+      {:values []})))
+
+(api/defendpoint POST "/:id/dimension"
+  "Sets the dimension for the given field at ID"
+  [id :as {{dimension-type :type dimension-name :name human_readable_field_id :human_readable_field_id} :body}]
```

Comment:
@camsaul You writing it up sounds good, I'll work on it as a separate thing after the in-flight stuff gets done